### ISSUE TITLE
optimize aroundReceive of ActorAdapter

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#27294]
we could classify the message into :
- `untyped.AutoReceivedMessage`
- `InternalMessage`
- `Signal`
- other messages

instead of current flatten pattern matching

**Comment by johanandren** [comment]
I actually benchmarked something like this (matching on `InternalMessage`) while optimizing the adapter and interpreter and it did not yield an improvement (probably hierarchy of types vs concrete type instanceof).

**Comment by hepin1989** [comment]
Yeah, because there are both message types from untyped and typed. Seems not easy to make it a switch table.
